README.md

```diff
@@ -30,236 +33,173 @@
-                    .addOption(OptionType.STRING, "reason", "The ban reason") // optional reason
-        ).queue();
-    }
-    
-    @Override
-    public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
-        // make sure we handle the right command
-        switch (event.getName()) {
-            case "ping":
-                long time = System.currentTimeMillis();
-                event.reply("Pong!").setEphemeral(true) // reply or acknowledge
-                     .flatMap(v ->
-                          event.getHook().editOriginalFormat("Pong: %d ms", System.currentTimeMillis() - time) // then edit original
-                     ).queue(); // Queue both reply and edit
-                break;
-            case "ban":
-                // double check permissions, don't trust Discord on this!
-                if (!event.getMember().hasPermission(Permission.BAN_MEMBERS)) {
-                    event.reply("You cannot ban members! Nice try ;)").setEphemeral(true).queue();
-                    break;
-                }
-                User target = event.getOption("user", OptionMapping::getUser);
-                // optionally check for member information
-                Member member = event.getOption("user", OptionMapping::getMember);
-                if (!event.getMember().canInteract(member)) {
-                    event.reply("You cannot ban this user.").setEphemeral(true).queue();
-                    break;
-                }
-                // Before starting our ban request, tell the user we received the command
-                // This sends a "Bot is thinking..." message which is later edited once we finished
-                event.deferReply().queue();
-                String reason = event.getOption("reason", OptionMapping::getAsString);
-                AuditableRestAction<Void> action = event.getGuild().ban(target, 0); // Start building our ban request
-                if (reason != null) // reason is optional
-                    action = action.reason(reason); // set the reason for the ban in the audit logs and ban log
-                action.queue(v -> {
-                    // Edit the thinking message with our response on success
-                    event.getHook().editOriginal("**" + target.getAsTag() + "** was banned by **" + event.getUser().getAsTag() + "**!").queue();
-                }, error -> {
-                    // Tell the user we encountered some error
-                    event.getHook().editOriginal("Some error occurred, try again!").queue();
-                    error.printStackTrace();
-                });
-                break;
-            default:
-                System.out.printf("Unknown command %s used by %#s%n", event.getName(), event.getUser());
-        }
+public class SlashCommandListener extends ListenerAdapter {
+  @Override
+  public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
+    switch (event.getName()) {
+      case "say" -> {
```

Comment:
Shouldn't this use the Java 8 syntax?

Reply:
No, this library can and should be used with 21.